packages/odata-service-writer/src/data/annotations.ts

```diff
@@ -13,15 +13,17 @@ import type { NamespaceAlias, OdataService } from '../types';
 export function getAnnotationNamespaces({ metadata, annotations }: Partial<OdataService>): NamespaceAlias[] {
     // Enhance service with annotations namespaces
     const schemaNamespaces = metadata ? getNamespaces(metadata) : [];
-
-    if (annotations?.xml) {
+    const edmxAnnotations = annotations as EdmxAnnotationsInfo;
```

Comment:
Should we not test for the property of `EdmxAnnotationsInfo` before casting to that type?

```suggestion
    if (edmxAnnotations?.xml) {
        const edmxAnnotations = annotations as EdmxAnnotationsInfo;
```

Reply:
Or better still change the parameter type to be `EdmxAnnotationInfo` if that is what should be passed to this function.